packages/contracts-bedrock/scripts/checks/test-validation/main.go

```diff
@@ -161,23 +161,13 @@ func checkTestStructure(artifact *solc.ForgeArtifact) []error {
 			// Pattern: <ContractName>_Harness
 			continue
 		} else if len(contractParts) == 3 && contractParts[2] == "Test" {
-			// Check for uncategorized test pattern
-			if contractParts[1] == "Uncategorized" || contractParts[1] == "Unclassified" {
-				// Pattern: <ContractName>_Uncategorized_Test
-				continue
-			} else {
-				// Pattern: <ContractName>_<FunctionName>_Test - validate function exists
-				functionName := contractParts[1]
-				if !checkFunctionExists(artifact, functionName) {
-					// Convert to camelCase for error message
-					camelCaseFunctionName := strings.ToLower(functionName[:1]) + functionName[1:]
-					errors = append(errors, fmt.Errorf("contract '%s': function '%s' does not exist in source contract", contractName, camelCaseFunctionName))
-				}
-			}
+			errors = append(errors, checkTestMethodName(artifact, contractName, contractParts[1], "")...)
 		} else if len(contractParts) == 3 && contractParts[2] == "Harness" {
 			// Pattern: <ContractName>_<Descriptor>_Harness
 			// (e.g., OPContractsManager_Upgrade_Harness)
 			continue
+		} else if len(contractParts) == 4 && contractParts[3] == "Test" {
+			errors = append(errors, checkTestMethodName(artifact, contractName, contractParts[1], contractParts[2])...)
```

Comment:
I'm being opinionated here and allowing an additional test name pattern: `ContractName_MethodName_FeatureNameEnabled_Test`.  Ideally we'd probably check that the feature name existed but I haven't actually implemented that here.